corehq/warehouse/etl.py

```diff
@@ -44,6 +46,14 @@ def load(cls, batch):
 
         assert issubclass(cls, WarehouseTable)
         database = db_for_read_write(cls)
+        if append:
+            batches = cls.objects.distinct('batch').values_list('batch', flat=True)
+            if batches:
+                oldest = Batch.objects.filter(pk__in=batches).order_by('start_datetime').first()
+                if batch.start_datetime < oldest.start_datetime:
+                    batch.end_datetime = oldest.start_datetime
+                else:
+                    return
```

Comment:
i'm not completely following what append is used for and what is happening when it's True - what's this for?

__in queries also don't perform that well at high scale